app/controllers/local_transaction_controller.rb

```diff
@@ -16,20 +16,32 @@ class LocalTransactionController < ApplicationController
   def search
     if request.post?
       @location_error = location_error
+
       if @location_error
         @postcode = postcode
-      elsif mapit_response.location_found? && lgsl == 364 && country_name == "Northern Ireland" ## LGSL code 364 = electoral registration
-        redirect_to local_transaction_results_path(local_authority_slug: "electoral-office-for-northern-ireland")
-      elsif mapit_response.location_found? && local_authority_slug
-        redirect_to local_transaction_results_path(local_authority_slug: local_authority_slug)
+      elsif mapit_response.location_found?
+        slug = if lgsl == 364 && country_name == "Northern Ireland"
+                 "electoral-office-for-northern-ireland"
+               else
+                 local_authority_slug
+               end
+
+        redirect_to local_transaction_results_path(local_authority_slug: slug)
       end
     end
   end
 
   def results
-    @postcode = postcode
+    @postcode = params[:postcode]
```

Comment:
why not?
```suggestion
    @postcode = postcode
```

Reply:
Ah, meant to fix that - apologies - fixed